### DISCUSSION TITLE
Whats the debug/ build process you use?

### DISCUSSION THREAD

**Discussion by RollsChris** [cmoog/vscode-sql-notebook#39] [Q&A]
i have been struggling with building the app and getting break points to hit.. I'm thinking I'm doing something wrong.

What process do you use? do you run "npm run build" much?

thanks

**Comment by cmoog** [comment]
I only use `npm run build` as a sanity check to test before merging something substantial.

During development, I use the [launch scripts](https://github.com/cmoog/vscode-sql-notebook/blob/master/.vscode/launch.json) designed for extension development. I don't use the debugger though, so there could certainly be something causing that to break. And since we moved from `tsc` to `@vercel/ncc` for bundling, I do think the problem matcher has been broken.

For reference: clicking the green icon here will launch a new vscode window that is connected to the extension development build.

<img width="332" alt="Screen Shot 2022-04-02 at 5 26 53 PM" src="https://user-images.githubusercontent.com/7585078/161403311-a92b1782-a705-45d7-aacd-692d985e891e.png">

**Comment by RollsChris** [comment]
Yea i'm not sure what's going on... i'm unable to continue development.. not sure what the need is for the all the file output changes and why your using vsce and ncc .. need to have a look. i dont feel we are doing anything fancy why change from the norm

**Reply by cmoog** [reply]
Every VS Code extension uses [vsce](https://www.npmjs.com/package/vsce) to bundle the final package into a vsix properly.

As for `ncc`, it is simply a wrapper around `tsc` that inlines the node_modules to decrease the file count in the final bundle, as I described in [this](https://github.com/cmoog/vscode-sql-notebook/commit/990ff6ce57df1b3b20903ecc8bdb3a240cc394c6) commit message. Everything about the build process is pretty standard, nothing fancy (with the exception of nix, but that is purely opt-in).

**Reply by RollsChris** [reply]
Can you try your end please? you seem to have more knowledge around this.

I can see in launch.json you use pwa-extensionhost and the default tutorial uses extensionhost? i tried changing it and didn't fix anything just curios to what it is.

**Comment by RollsChris** [comment]
Yes thats what I'm doing. I have had it working before.. not sure what's going on. Might be something I have changed but can't even see any errors

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
